pkg/virt-controller/services/template.go

```diff
@@ -526,21 +526,19 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 		})
 	}
 
+	vcpus := hardware.GetNumberOfVCPUs(vmi.Spec.Domain.CPU)
+
+	if vcpus != 0 {
+		resources.Requests[k8sv1.ResourceCPU] = *resource.NewQuantity(vcpus, resource.BinarySI)
```

Comment:
this effectively prevents the possibility of overcommit, which we want users to have the ability to do.

Reply:
The API allows overcommit indeed, but it gets complicated to configure a simple VM this way as you have to specify both resources and topology in certain cases.

What we were trying to discuss here is a way to make sure a very simple VM gets the requested resources by default with minimal configuration.

Allowing override of the defaults is a must of course and should not be broken.